modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockStorage.scala

```diff
@@ -94,17 +102,18 @@ class BlockStorage[F[_]: Sync: Random](blocks: MapRef[F, ProofsHash, Option[Stor
     def removeObsoleteBlocks: F[Unit] =
       obsoleteToRemove.toList.traverse { hash =>
         blocks(hash).modify {
-          case Some(WaitingBlock(_)) => (None, ().asRight)
-          case other                 => (other, UnexpectedBlockStateWhenRemoving(hash, other).asLeft)
+          case Some(WaitingBlock(_)) | Some(PostponedBlock(_)) => (None, ().asRight)
+          case other                                           => (other, UnexpectedBlockStateWhenRemoving(hash, other).asLeft)
         }.flatMap(_.liftTo[F])
       }.void
 
     def resetBlocks: F[Unit] =
       toReset.toList.traverse { hash =>
         blocks(hash).modify {
-          case Some(AcceptedBlock(block))  => (WaitingBlock(block.signed).some, ().asRight)
-          case Some(waiting: WaitingBlock) => (waiting.some, ().asRight)
-          case other                       => (None, UnexpectedBlockStateWhenResetting(hash, other).asLeft)
+          case Some(AcceptedBlock(block))      => (WaitingBlock(block.signed).some, ().asRight)
+          case Some(waiting: WaitingBlock)     => (waiting.some, ().asRight)
+          case Some(postponed: PostponedBlock) => (WaitingBlock(postponed.block).some, ().asRight)
```

Comment:
so `reset` is the moment when postponed block goes back to waiting, right @kriscieslak?